Upgrade to dash 3.0.0 for GUI

benrichard-amd opened the pull request:
`requirements.txt` listed `dash >= 1.12.0`, but as of `3.0.0` our code is no longer compatible. `app.run_server()` has been replaced by `app.run()`, and integer indexing with `[]` has been deprecated.

This PR updates `requirements.txt` to `dash >= 3.0.0` and updates the code accordingly.

benrichard-amd committed:
Update GUI code to be compatible with dash 3.0.0

benrichard-amd committed:
Fix deprecating integer indexing when using dash

Noted in warning output, Dash prefers to use iloc()
for integer indexing. Using [] is deprecated and
will be removed in the future.

benrichard-amd committed:
Fix formatting

ywang103-amd committed:
TCC per channel metric's fix for rocprofv3 (#597)

ywang103-amd committed:
Revert "fix the error of output path of multi-node mode (#616)" (#621)

xuchen-amd committed:
Improve chip info logic. (#581)

* Clean up unused functions.

* Fix number of XCDs for MI300X CPX (core partition).

* Add support for memory partition mode.

* Modify total_xcd to adapt to all gpu models.

* Run black and isort.

* Make gpu_arch regex more generic.

* Add error checking for compute partition mode num xcds.

* Set gpu_chip_id as optional.

* Fix get_gpu_model.

---------

Signed-off-by: xuchen-amd <xuchen@amd.com>

vedithal-amd commented:
@benrichard-amd  were you able to test the standalone gui? were you able to confirm there is no UI changes to the way our gui looks right now due to using dash 3?

cfallows-amd committed:
Add Alibaba Cloud Linux 3 to distro checking for roofline (#620)

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

dependabot[bot] committed:
Bump rocm-docs-core from 1.17.0 to 1.18.1 in /docs/sphinx (#611)

Bumps [rocm-docs-core](https://github.com/ROCm/rocm-docs-core) from 1.17.0 to 1.18.1.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.17.0...v1.18.1)

---
updated-dependencies:
- dependency-name: rocm-docs-core
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ywang103-amd committed:
fix the part that casue _sum counters missing for TCC (#627)

vedithal-amd committed:
Remove static pmc files (#606)

* Delete static pmc files

* Counter parsing changes
	- Move counter parsing logic to another function
	- Fix counter parsing regex
	- Log list of counters being collected

* Sanity check counters supported by rocprof
	- Emit warning instead of error since rocprof support counters
	  list might be inaccurate

* Do not collect these counters
	- TCP_TCP_LATENCY_sum (except for gfx908 and gfx90a)
	- SQC_DCACHE_INFLIGHT_LEVEL

* Update logic of writing TCC channel counter definition yaml file

* Fix bug in capture_subprocess_output() utility function
	- Make logging optional in capture_subprocess_output()

* Fix formatting and tests

* Update changelong

cfallows-amd committed:
Datatype selection option for roofline (#624)

Added command line option to specify which datatype(s) to capture into the roofline PDF(s).
All datatypes are still collected by roofline call if applicable, but only specific datatypes are plotted into PDF outputs. Will dump out all datatypes into one graph, but separate FP from Int into two graphs if needed. Will skip datatype and give error message if the datatype is not valid on a particular gpu arch.
Default is FP32

Reworked roofline calls and plotting to be general enough such that any new datatypes added into rocm-amdgpu-bench can easily be reflected in rocprof-compute with simple modifications in roofline_calc.py.

Adjusted ctest to reflect expected default pdf outputs from roofline.

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

ywang103-amd committed:
disable TCC flattern for rocprofv1 to avoid its crash becasue of unsupported implementation (#629)

ywang103-amd committed:
fix the crash related to agent id in rocprofv3 (#631)

coleramos425 committed:
Generalize locale checker to support more UTF-8 types (#623)

Signed-off-by: coleramos425 <colramos@amd.com>

xuchen-amd committed:
Improve readability. (#628)

coleramos425 committed:
Fix incorrect logging in `mi_gpu_spec.py` (#626)

* Move console logging to logger function to avoid circular dependency in utils module

Signed-off-by: coleramos425 <colramos@amd.com>

* Apply python formatting

Signed-off-by: coleramos425 <colramos@amd.com>

* Remove the default StreamHandler before adding the custom

 If you are not explicitly removing this default handler, it could be causing duplicate outputs.

Signed-off-by: coleramos425 <colramos@amd.com>

* Fix lingering bugs from merge conflict resolution

Signed-off-by: coleramos425 <colramos@amd.com>

* Comply to python formatting and update pre-commit hook helper

Signed-off-by: coleramos425 <colramos@amd.com>

* Removing redundant console_log call as the get_mi300_num_xcds() call, otherwise ALL Mi200 profiling runs will print this message

Signed-off-by: coleramos425 <colramos@amd.com>

---------

Signed-off-by: coleramos425 <colramos@amd.com>

cfallows-amd committed:
Add datatypes for roofline profiling (#642)

Rebuild of rocm-amdgpu-bench roofline binaries for MI200/MI300 systems with rocm6.
Added datatype options to roofline feature.

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

vedithal-amd committed:
Inject SQ_ACCUM_PREV_HIRES for LEVEL counters only (#641)

ywang103-amd committed:
fix ip block test by changing ways of extracting agent id (#639)

ywang103-amd committed:
fix the wrong number of channels of TCC counters to put in pmc txt file (#633)

benrichard-amd committed:
Read Accum_VGPR_Count from rocprof output if provided (#645)

feizheng10 committed:
Support host-trap PC Sampling on CLI (beta version)

ywang103-amd committed:
re-write fucntion that detects whether v1 is in use to avoid false negative result when ROCPROF is not set (#647)

xuchen-amd committed:
Improve chip id logic (#648)

* Improve chip id logic, add missing physical and virtual chip ids.

vedithal-amd committed:
Weekly rebase liangdin-test on top of amd-mainline (#650)

raramakr committed:
SWDEV-521636 - Add dependent script path to system path in rocprof-compute (#651)

In wheel environment, rocprof-compute in bin folder is not a soft link. For executing rocprof-compute from bin folder, the system path should also have the dependency script paths. Added the same

xuchen-amd committed:
Enable tuned performance counters for gfx950 (#652)

* Enable non-functional performance counters for gfx950.

* Update changelog.

* Add none value metrics for non-gfx950 socs

* Remove rocprofv3 missing metrics.

xuchen-amd committed:
remove flask debug msg (#655)

* Suppress Flask warning message in quiet mode.

* Init args.gui if dne.

xuchen-amd committed:
Add mi300 TCP counter tests (#644)

* Add new sample applications.

* Generalize py test launcher for additional apps.

* Add TCP pytest, and add to ctest.

* Update licensing.

* Disable for non-mi300 machines.

vedithal-amd commented:
@benrichard-amd can you please merge this, thanks!

xuchen-amd committed:
Add mi350 ta td tcp tcc counters (#653)

* Add mi350 TA and TD metrics.

* Add mi350 TCC metrics, and separate write and atomic metrics.

* Add mi350 TCP metrics.

* Add none values for non-gfx950 socs, remove missing metrics in rocprofv3.

---------

Signed-off-by: xuchen-amd <xuchen@amd.com>

vedithal-amd committed:
Support MI 350 profiling (#632)

* Add MI 350 hardware information

* Refactor MI GPU YAML file and corresponding interface

* Add SoC file for gfx950 architecture

* Add analysis report configs for MI 350 containing existing metrics

* Add placeholder None valued metrics for previous architectures to make
  baseline comparison work

* Enable testing on MI 350

* Analysis config metric changes
    - SPI changes
        - Update metric formula for default SPI pipe counter
             - Use efficiently collected pipe wise SPI counters
        - Add SPI Wave Occupancy
        - Add Scheduler-Pipe Wave Utilization
        - Update formula for VGPR Writes
        - Add Scheduler-Pipe FIFO Full Rate
   - CPC changes
	- Add CPC SYNC FIFO Full Rate
	- Add CPC CANE Stall Rate
        - Add CPC ADC Utilization
   - SQ changes
        - Add VALU co-issue efficiency
        - Add F6F4 datatype metrics
        - Update formula for total FLOPs by adding F6F4 counters
        - Add LDS STORE / LOAD / ATOMIC metrics
        - Add LDS STORE / LOAD / ATOMIC bandwidth
        - Add LDS FIFO and TA ADDR / CMD / DATA FIFO full rates

* Collect TCP_TCP_LATENCY_sum only for gfx950 (MI 350)

* Do not inject SQ_ACCUM_PREV_HIRES unnecesarily

* Do not hardcode memory and shader clock speeds

* Write num_hbm_channels to sysinfo.csv instead of hbm_bw while profiling

* Move generate sysinfo.csv to pre processing step of profiling

* Add warnings to use --specs-correction for missing sysinfo.csv values during analysis phase

* Update CHANGELOG

* Analysis phase warning to use --specs-correction when needed

cfallows-amd committed:
Fixes for roofline datatype plot outputs (#659)

Profile mode:
Fix roofline plots for datatypes that have peakVALU only. Check for highest roofline to plot the bandwidth lines to proper height, don't rely on existence of peakMFMA for every datatype.
Analyze mode:
Add roofline-data-type option for viewing pdfs in standalone gui. Default is same as profile mode, FP32.

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

xuchen-amd committed:
Add mi325 specs. (#663)

prbasyal-amd committed:
Broken link and reference text updated (#664)

dependabot[bot] committed:
Bump rocm-docs-core from 1.18.1 to 1.18.2 in /docs/sphinx (#657)

Bumps [rocm-docs-core](https://github.com/ROCm/rocm-docs-core) from 1.18.1 to 1.18.2.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.18.1...v1.18.2)

---
updated-dependencies:
- dependency-name: rocm-docs-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

danielsu-amd committed:
External CI: add parallel mainline checks for develop and staging branches (#666)

cfallows-amd committed:
Fix rpath checks during RPM generation on RHEL10 (#669)

Invalid rpath on roofline binaries reported during build testing for new RHEL10 addition, removed rpaths to prevent rpath check failures.

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

cfallows-amd committed:
Add roofline support for rhel10 (#667)

-add check for rhel10 (platform:el10), force use rhel roof binary
-update changelog in 'unreleased- added' section

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

ywang103-amd committed:
configure rocprofv3 as default for unit test (#668)

ywang103-amd committed:
change default rocprof version to v3 when not setting env variable (#673)

dgaliffiAMD committed:
Bump VERSION to 3.2.0

cfallows-amd committed:
Update runner distro in Formatting workflow (#678)

Update formatting workflow to use 22.04. 20.04 deprecated last week.

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

xuchen-amd committed:
Patch in new rocprofv3 metrics. (#679)

xuchen-amd committed:
process hip trace output. (#654)

Signed-off-by: xuchen-amd <xuchen@amd.com>

xuchen-amd committed:
Add test for gfx942 number of xcds. (#674)

* Add test for 9fx942 number of xcds.

* Improve the structure of mi gpu specs, add num_xcds_spec_class test.

* Add to ctest.

---------

Signed-off-by: xuchen-amd <xuchen@amd.com>

danielsu-amd committed:
Set rocprofV3 agent-index to `absolute` (#675)

Signed-off-by: Daniel Su <danielsu@amd.com>

cfallows-amd committed:
Update CODEOWNERS (#680)

Add rp-compute technical writer directly for any documentation review.
Remove existing packaging review requests for single user; every repo owner should be notified.

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

cfallows-amd committed:
Enable roofline for MI350 series (#677)

Rework of roofline binaries generated from rocm-amdgpu-bench
- removed arch identifier in bin name
- removed rocm5 bins altogether

Updated required distros for roofline
- updated distro checks and bin naming
- moved up ubuntu20.04->22.04 and sles15.3->15.6 per rocm support

Enabled ctests for mi350 for test_roof_*
- removed mi350 series check to skip these specific tests

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

benrichard-amd committed:
Avoid crash when profiling data not generated (#694)

* Avoid crash when profiling data not generated

-Handle case where program has no kernel launches
-Improve error messages
-Avoid roofline when profiling data is missing

Signed-off-by: benrichard-amd <ben.richard@amd.com>

* Update other soc_gfx files to catch missing pmc_perf.csv

* Fix formatting

* Fix incorrectly ordered imports

---------

Signed-off-by: benrichard-amd <ben.richard@amd.com>

vedithal-amd committed:
Fix PC sampling analysis config issue (#697)

vedithal-amd committed:
Use gpu model series instead of gpu model name for testing (#696)

xuchen-amd committed:
Improve gpu spec tests using chip ids. (#701)

cfallows-amd committed:
Fix setting roofline-data-type option in both profile and analyze modes (#702)

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

vedithal-amd committed:
Implement interface to rocprofiler sdk (#695)

* Setting ROCPROF=rocprofiler-sdk environment variable will use rocprofiler-sdk C++ library instead of rocprofv3 python script

* Add runtime option --rocprofiler-sdk-library-path to use custom version of rocprofiler sdk library
    * Add --rocprofiler-sdk-library-path conftest option for tests

* Setup appropriate environment variables to inject rocprofiler sdk code to user command
    * Add env. vars. for counter collection and filtering
    * Add env. vars. for pc sampling

* Use python bindings to list counters supported by rocprofiler sdk

vedithal-amd committed:
Add James Siddley to code reviewers

vedithal-amd committed:
Fix MI350 tests (#706)

- TCC counter collection tests are still failing due to recent
  rocprofiler-sdk change

cfallows-amd committed:
Check mode during soc init for roofline (#705)

Check mode before creating roofline object- skip if only printing specs

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

benrichard-amd committed:
Update illegal character check for profile name (#703)

jamessiddeley-amd committed:
Fixed duplicate keys in analysis_configs yamls (#707)

* fixed duplicate keys in analysis_configs yamls

* Fix: removed TODO comment

Signed-off-by: James Siddeley <james.siddeley@amd.com>

---------

Signed-off-by: James Siddeley <james.siddeley@amd.com>

cfallows-amd committed:
Add F4 F6 to roofline for MI350 series (#709)

Add roofline bins with FP4 FP6 datatypes enabled for gfx950 arch

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

benrichard-amd committed:
Update CHANGELOG

benrichard-amd commented:
PR is stale and I messed up my branch trying to rebase. Will submit a new PR.
